armi/materials/material.py

```diff
@@ -88,15 +85,25 @@ def __init__(self):
         self.theoreticalDensityFrac = 1.0
         self.cached = {}
         self._backupCache = None
+        self._name = self.__class__.__name__
 
         # call subclass implementations
         self.setDefaultMassFracs()
 
     def __repr__(self):
-        return "<Material: {0}>".format(self.getName())
+        return f"<Material: {self._name}>"
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, nomen):
```

Comment:
If someone defines a custom name that isn't the class name, how do you intend for this to function with regards to `resolveMaterialClassByName()`?

Reply:
I second this question. Or maybe you could add a unit test for `resolveMaterialClassByName` that covers the custom name case?

Reply:
Right now, every material class in armi can be instantiated by name. In the case of custom materials it is "Custom". 
![image](https://github.com/terrapower/armi/assets/103008519/93e80ff7-7331-4435-a804-49cc4852c710)

If you assume, the user is going to rename the name on "Custom" materials to something useful, then sure it doesn't work for that right now. At least that is isolated to one material class that's an odd child in it's usage. By adding this method onto the base material class, you are encouraging people to rename any material class as an acceptable behavior. When that starts happening, then `resolveMaterialClassByName()` stops working for a larger number of classes. 
![image](https://github.com/terrapower/armi/assets/103008519/ca7bf042-2f97-4ff4-b9b4-94410db40188)

Reply:
Maybe we could just shift this setter method to the custom material class from the base material class and call it good?

Reply:
Yes, I was suggesting shifting it from the base material class like it is in this PR to the Custom material class (https://github.com/terrapower/armi/blob/main/armi/materials/custom.py). That would provide some contextual clues to a user that they shouldn't change the name of a non-custom material.

Reply:
Okay, so here are some examples in the docs of `Custom` materials being used:

https://terrapower.github.io/armi/tutorials/walkthrough_lwr_inputs.html#setting-up-the-blueprints
https://terrapower.github.io/armi/user/inputs/blueprints.html#custom-isotopics

The important thing to note here is the name of every custom material is "Custom".  And then they are given specific isotopics.

I have checked with in-house TerraPower users of Custom materials, and this appears to be the only case.

As such, I believe the `resolveMaterialClassByName()` function does not need to be changed to work with this PR.